source/blueprints/banking.rst

```diff
@@ -0,0 +1,546 @@
+                    responder.reply(['Ok we have scheduled your credit card payment for your {payment} of ${total_balance}'])
+                    _put('credit', 0)
+		    _put('checking', get('checking') - responder.slots['total_balance'])
+	    #handle the system entity of the exact dollar amount they want to pay off
+            else:
+                responder.slots['amount'] = entity['value'][0]['value']
+                responder.reply(['Ok we have scheduled your credit card payment for {amount}'])
+                _put('credit', _get('credit') - entity['value'][0]['value'])
+		_put('checking', get('checking') - responder.slots['amount'])
+	responder.listen() 
+
+
+
+For a more realistic implementation of ``pay_creditcard`` that deals with varied user flows and the full code behind all the dialogue state handlers, see the ``__init__.py`` file in the blueprint folder.
+
+
+5. Slot Filling
+^^^^^^^^^^^^^^^
+
+Slot filling logic allows you to easily request for a missing entity. You can set custom responses in the slot filling form to prompt the user with when an entity is missing in the user request. Once the missing entities in the form have been provided you can handle the rest of the logic as you would in the handler function of your intent. More detailed information on slot filling is below along with an example from the banking assistant. 
+
+Removes the need to provide @app.handle if @app.auto_fill is being called.
+The handler functionality will be called by the slot filling class once that process is completed.
+
+Form is a dictionary containing the following entries:
+
+  - ``entities`` - (list, required): List of FormEntity objects with each defined for one entity slot to be filled.
+  - ``max_retries`` - (int, optional, default 1): maximum number of retries allowed per entity or slot if user response is invalid.
+  - ``exit_msg`` - (str, optional): If slot filling is exited abruptly without completion, define custom message to display.
+  - ``exit_keys`` - (list, optional): List of exit hints for the slot filling flow. If these words or phrases are said by the user, the slot filling logic exits. Default: ['cancel', 'restart', 'exit', 'reset'].
+
+FormEntity is a class that allows creation of entity objects for slot filling and comprises of the following attributes:
+
+  - ``entity``  - (str, required): Entity name
+  - ``role`` - (str, optional): The role of the entity
+  - ``responses`` - (list or str, optional): message for prompting the user for missing entities
+  - ``retry_response`` - (str, optional): message for re-prompting users. If not provided, defaults to responses.
+  - ``value`` - (str, optional): The resolved value of the entity
+  - ``default_eval`` - (bool, optional): Use system validation (default: True)
+  - ``hints`` - (list, optional): Developer defined list of keywords to verify theuser input against
+  - ``custom_eval`` - (func, optional): custom validation function (should return bool:validated or not)
```

Comment:
@abhijeetsidhu  @ritvikshrivastava I don't think we need to repeat the documentation that's already in the user guide. It would be better to just link to the appropriate section. Restrict the content in this section to just the code example from the banking blueprint.